src/main/java/seedu/club/logic/commands/SignUpCommand.java

```
@@ -0,0 +1,65 @@
+import static seedu.club.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.club.logic.parser.CliSyntax.PREFIX_PASSWORD;
+import static seedu.club.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.club.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.club.logic.parser.CliSyntax.PREFIX_USERNAME;
+
+import seedu.club.logic.CommandHistory;
+import seedu.club.logic.UndoRedoStack;
+import seedu.club.model.Model;
+import seedu.club.model.member.Member;
+
+//@@author Song Weiyang
+/**
+ * Signs up a member when the clubbook is empty
+ */
+public class SignUpCommand extends Command {
+    public static final String COMMAND_WORD = "signup";
+
+    public static final String COMMAND_FORMAT = "login n/ p/ e/ m/ [pic/ ] ";
+
+    public static final String MESSAGE_SUCCESS = "Sign up successful! Please log in again";
+    public static final String MESSAGE_FAILURE = "There are already members in the clubbook. Log In to continue";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sign up a member to ClubConnect. "
+            + "Parameters: "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_PHONE + "PHONE "
+            + PREFIX_EMAIL + "EMAIL "
+            + PREFIX_MATRIC_NUMBER + "MATRIC NUMBER "
+            + "[" + PREFIX_GROUP + "GROUP] "
+            + "[" + PREFIX_TAG + "TAG]... "
+            + PREFIX_USERNAME + "username "
+            + PREFIX_PASSWORD + "password\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_PHONE + "98765432 "
+            + PREFIX_EMAIL + "johnd@example.com "
+            + PREFIX_MATRIC_NUMBER + "A0123456H "
+            + PREFIX_TAG + "friends "
+            + PREFIX_TAG + "owesMoney ";
+    private final Member toSignUp;
+
+    public SignUpCommand(Member member) {
+        this.toSignUp = member;
+    }
+
+    @Override
+    public CommandResult execute() {
+        if (model.getClubBook().getMemberList().isEmpty()) {
```

Would it be better to propagate an exception instead?